great_expectations/checkpoint/checkpoint.py

```diff
@@ -158,13 +166,13 @@ def run(
         if not self.validation_definitions:
             raise CheckpointRunWithoutValidationDefinitionError()
 
-        added, errors = self.is_added()
-        if not added:
+        diagnostics = self.is_added()
+        if not diagnostics.is_added:
             # The checkpoint itself is not added but all children are - we can add it for the user
-            if len(errors) == 1 and isinstance(errors[0], CheckpointNotAddedError):
+            if not diagnostics.parent_added and diagnostics.children_added:
                 self._add_to_store()
             else:
-                raise CheckpointRelatedResourcesNotAddedError(errors=errors)
+                diagnostics.raise_for_error()
```

Comment:
@cdkini this line being in a conditional is the reason I was surprised below that it didn't always raise. Not saying anything needs to change. Just thinking aloud.